scripts/evaluate.py

```diff
@@ -140,27 +141,53 @@ def test_website(args, checkpoint_file, experiment_name):
             )
 
 
-def worker(args, checkpoint_file, experiment_name):
+def worker(results_dir, manifests_dir, reward_func, bandwidth, cpu_slowdown, latency, checkpoint_file, experiment_name):
     try:
-        if website_exists(experiment_name, args.results_dir):
+        if website_exists(experiment_name, results_dir):
             print(f"evaluating {checkpoint_file} ... skipping (already exists)")
             return
         print(f"evaluating {checkpoint_file} ...")
-        test_website(args, checkpoint_file, experiment_name)
+        test_website(results_dir, manifests_dir, reward_func, bandwidth, cpu_slowdown, latency, checkpoint_file, experiment_name)
     except KeyboardInterrupt:
         raise
     except:
         traceback.print_exc()
 
+def generate_random_params():
+    # {12, 14, 16, 18, 20, 22, 24} * {20, 40, 60, 80, 100} * {2x, 4x}
+    while True:
+        for cpu in range(2):
+            for latency in range(5):
+                for bandwidth in range(7):
+                    chosen_cpu = (cpu + 1) * 2
+                    chosen_latency = (latency + 1) * 20
+                    chosen_bandwidth = 12 + (bandwidth * 2)
+                    yield (chosen_bandwidth, chosen_cpu, chosen_latency)
 
 def main(args):
-    websites = list(get_test_websites(args.ray_results_dir))
-    with concurrent.futures.ThreadPoolExecutor(max_workers=args.num_workers) as pool:
-        f = [
-            pool.submit(worker, args, checkpoint_file, experiment_name) for checkpoint_file, experiment_name in websites
-        ]
-        print("Finished processing", len([p.result() for p in f]), "websites")
+    if (not args.randomize) and (args.bandwidth is None or args.cpu_slowdown is None or args.latency is None):
+        print("Either choose randomize option or provide all three of (bw, latency, cpu).")
+    else:
+        websites = list(get_test_websites(args.ray_results_dir))
+        with concurrent.futures.ThreadPoolExecutor(max_workers=args.num_workers) as pool:
+            random_params = generate_random_params()
+            for checkpoint_file, experiment_name in websites:
+                for i in range(70):
+                    print("starting permutation ", i, " for website ", checkpoint_file, experiment_name)
+                    results_dir = args.results_dir
+                    manifests_dir = args.manifests_dir
+                    reward_func = args.reward_func
+                    if args.randomize:
+                        bandwidth, cpu_slowdown, latency = random_params.__next__()
+                    else:
+                        bandwidth = 0
+                        cpu_slowdown = 0
+                        latency = 0
```

Comment:
shouldn't these be `args.bandwidth`, `args.cpu_slowdown`, `args.latency` respectively